src/soundsourceproxy.cpp

```diff
@@ -326,50 +292,170 @@ void SoundSourceProxy::nextSoundSourceProvider() {
 }
 
 void SoundSourceProxy::initSoundSource() {
-    DEBUG_ASSERT(!m_pSoundSource);
-    DEBUG_ASSERT(!m_pAudioSource);
-    while (!m_pSoundSource) {
+    DEBUG_ASSERT(m_pSoundSource.isNull());
+    DEBUG_ASSERT(m_pAudioSource.isNull());
+    QString trackType;
+    if (!m_pTrack.isNull()) {
+        trackType = m_pTrack->getType();
+    }
+    while (m_pSoundSource.isNull()) {
         Mixxx::SoundSourceProviderPointer pProvider(getSoundSourceProvider());
-        if (!pProvider) {
-            qWarning() << "Failed to obtain SoundSource for file"
-                    << getFilePath();
-            return; // failure -> exit loop
+        if (pProvider.isNull()) {
+            qWarning() << "No SoundSourceProvider for file"
+                    << getUrl();
+            // Failure -> exit loop
+            break;
         }
         m_pSoundSource = pProvider->newSoundSource(m_url);
-        if (m_pSoundSource) {
-            qDebug() << "Obtained SoundSource for"
-                    << getFilePath()
-                    << "from provider"
-                    << pProvider->getName();
-            return; // success -> exit loop
+        if (m_pSoundSource.isNull()) {
+            qWarning() << "SoundSourceProvider"
+                    << pProvider->getName()
+                    << "failed to create a SoundSource for file"
+                    << getUrl();
+            // Switch to next provider...
+            nextSoundSourceProvider();
+            // ...and continue loop
+            DEBUG_ASSERT(m_pSoundSource.isNull());
         } else {
-            qWarning() << "Failed to obtain SoundSource for file"
-                    << getFilePath()
-                    << "from provider"
-                    << pProvider->getName();
+            trackType = m_pSoundSource->getType();
+            qDebug() << "SoundSourceProvider"
+                    << pProvider->getName()
+                    << "created a SoundSource for file"
+                    << getUrl()
+                    << "of type"
+                    << trackType;
+            // Success -> exit loop
+            break;
         }
-        // Continue with next SoundSource provider
-        nextSoundSourceProvider();
     }
+    if (!m_pTrack.isNull()) {
+        m_pTrack->setType(trackType);
```

Comment:
since we "break" after setting trackType above, we can move this just before the break and remove the m_pTrack->getType() call above.